src/main/java/de/hysky/skyblocker/skyblock/item/background/ItemBackgroundManager.java

```diff
@@ -0,0 +1,59 @@
+package de.hysky.skyblocker.skyblock.item.background;
+
+import de.hysky.skyblocker.annotations.Init;
+import de.hysky.skyblocker.skyblock.item.background.adders.ItemRarityBackground;
+import de.hysky.skyblocker.skyblock.item.background.adders.JacobMedalBackground;
+import de.hysky.skyblocker.utils.scheduler.Scheduler;
+import net.fabricmc.fabric.api.client.screen.v1.ScreenEvents;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.item.ItemStack;
+
+import java.util.List;
+
+public class ItemBackgroundManager {
+
+	private static final List<ColoredItemBackground<?>> BACKGROUNDS = List.of(
+			new ItemRarityBackground(),
+			new JacobMedalBackground()
+	);
+
+	@Init
+	public static void init() {
+		// Clear the cache of every background every 5 minutes
+		for (ColoredItemBackground<?> BACKGROUND : BACKGROUNDS) {
+			Scheduler.INSTANCE.scheduleCyclic(BACKGROUND.getCacheClearRunnable(), 6000);
+		}
+
+		// Hook into screen changes for per-background logic
+		ScreenEvents.BEFORE_INIT.register((client, screen, width, height) -> {
+			String title = screen.getTitle().getString();
+
+			for (ColoredItemBackground<?> background : BACKGROUNDS) {
+				background.onScreenChange(title, screen);
+			}
+
+			// Clear cache on screen close
+			ScreenEvents.remove(screen).register(s -> {
+				for (ColoredItemBackground<?> background : BACKGROUNDS) {
+					background.clearCache();
+				}
+			});
```

Comment:
I believe this is redundant in the presence of `ItemRarityBackground#onScreenChange`. They both clear the cache, but `ItemRarityBackground` does it conditionally.